### DISCUSSION TITLE
How to make react-shard work with react-hook-form

### DISCUSSION THREAD

**Discussion by riancz** [react-hook-form/react-hook-form#2525] [Q&A] [answered]
Hi guys,
I am using [**react-shards UI library**](https://github.com/DesignRevision/shards-react) (Its a wrapper over Bootstrap 4) on my project and I want to make **FormCheckbox** and **FormRadio** work with the react-hook-form.

Simply use `ref={register}` didn't work, so I tried to wrap it with **Controller component** like so:
```
<Controller
   as={<FormCheckbox>
      Men
  </FormCheckbox>}
  rules={{ required: true }}
  name='men'
  type='checkbox'
  control={control}
/>
```
This adds "value" attribute to the input tag but doesn't change "checked" property. Also the value in react-hook-form **changes from undefined to empty string**.

Any idea how to make this work? Thanks :)

**Comment by bluebill1049** [comment, marked as answer]
Do you have a codesandbox? 

Try `render` props with Controller as well.

**Reply by riancz** [reply]
Thanks @bluebill1049 for your quick response,
I made [this little codesandbox](https://codesandbox.io/s/festive-mcclintock-dj79n?file=/src/App.js) to showcase the problem. Look at how the checkboxes act when you click on the box vs on the label.

**Reply by bluebill1049** [reply]
it's a bug with react-shared: https://codesandbox.io/s/pedantic-jackson-lc6is?file=/src/App.js

click on the checkbox and label, behave differently

**Reply by riancz** [reply]
Oh the `onChange(e => onChange(e.target.checked))` helped. Thank you. I am trying to figure out what could be the problem with clicking on the checkbox and I thought it could be that the label containing the fake checkbox is missing `for` attribute so I tweaked a little their code to add it there but seems like this didn't help :( 

Than you again :)
